core/xim/index/IndexStore.lua

```diff
@@ -100,16 +98,26 @@ function _build_index_data(repodir)
                 cprint("[xlings:xim]: ${yellow}package file error: %s", file)
             end
         else
-            for version, _ in pairs(pkg.xpm[os_info.name]) do
+            local os_key = pkg.xpm[os_info.name].ref or os_info.name
+            for version, _ in pairs(pkg.xpm[os_key]) do
 
                 if version ~= "deps" then
                     local key = string.format("%s@%s", name_maintainer, version)
-                    index[key] = {
-                        version = version,
-                        installed = false,
-                        path = file
-                    }
+
+                    if pkg.xpm[os_key][version].ref then
+                        index[key] = {
+                            ref = name_maintainer .. "@" .. pkg.xpm[os_key][version].ref
+                        }
+                    else 
+                        index[key] = {
+                            version = version,
+                            installed = false,
+                            path = file
+                        }
+                    end
+
                     if version == "latest" then
+                        print(name_maintainer)
                         index[name_maintainer] = {
                             ref = key
                         }
```

Comment:
```lua
if not pkg.xpm or not pkg.xpm[os_info.name] then
    local local_pm = utils.local_package_manager()
    if not pkg.pm_wrapper or not pkg.pm_wrapper[local_pm] then
        cprint("[xlings:xim]: ${yellow}package file error: %s", file)
        return
    end
    local index_key = string.format("%s@%s", name_maintainer, local_pm)
    index[name_maintainer] = { ref = index_key }
    index[index_key] = {
        version = local_pm,
        installed = false,
        path = file
    }
    return
end
local os_key = pkg.xpm[os_info.name].ref or os_info.name
for version, _ in pairs(pkg.xpm[os_key]) do
      if version == "deps" then continue end
      local key = string.format("%s@%s", name_maintainer, version)
        local key = string.format("%s@%s", name_maintainer, version)
        if pkg.xpm[os_key][version].ref then
            index[key] = {
                ref = name_maintainer .. "@" .. pkg.xpm[os_key][version].ref
            }
        else
            index[key] = {
                version = version,
                installed = false,
                path = file
            }
        end
        if version == "latest" then
            index[name_maintainer] = {
                ref = key
            }
        end
    end
end
```

Reply:
Here the separation of local package manager and xpm gradually reform the index store. The difficulty of manage package independence will be potent. Actually the practical problem will occur in further development, I guess...messy.